exchanges/bybit/bybit_test.go

```diff
@@ -3893,3 +3886,89 @@ func TestTransformSymbol(t *testing.T) {
 		})
 	}
 }
+
+func TestWswsProcessOrder(t *testing.T) {
+	t.Parallel()
+	b := new(Bybit) //nolint:govet // Intentional shadow to avoid future copy/paste mistakes
+	require.NoError(t, testexch.Setup(b))
+
+	var wsOrderResponse WebsocketResponse
+	err := b.wsProcessOrder(&wsOrderResponse)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "unexpected end of JSON input")
+
+	wsOrderResponse = WebsocketResponse{
+		Data: []byte(`[{"category":"linear","symbol":"BTCUSDT","orderId":"c1956690-b731-4191-97c0-94b00422231b","orderLinkId":"","blockTradeId":"","side":"Sell","positionIdx":0,"orderStatus":"Filled","cancelType":"UNKNOWN","rejectReason":"EC_NoError","timeInForce":"IOC","isLeverage":"","price":"4.033","qty":"1.7","avgPrice":"4.24","leavesQty":"0","leavesValue":"0","cumExecQty":"1.7","cumExecValue":"7.2086","cumExecFee":"0.00288344","orderType":"Market","stopOrderType":"","orderIv":"","triggerPrice":"","takeProfit":"","stopLoss":"","triggerBy":"","tpTriggerBy":"","slTriggerBy":"","triggerDirection":0,"placeType":"","lastPriceOnCreated":"4.245","closeOnTrigger":false,"reduceOnly":false,"smpGroup":0,"smpType":"None","smpOrderId":"","slLimitPrice":"0","tpLimitPrice":"0","tpslMode":"UNKNOWN","createType":"CreateByUser","marketUnit":"","createdTime":"1733778525913","updatedTime":"1733778525917","feeCurrency":"","closedPnl":"0"}]`),
+	}
+
+	var o []order.Detail
+	var wg sync.WaitGroup
+	var ok bool
+	wg.Add(1)
+	go func() {
+		data := <-b.Websocket.DataHandler
+		o, ok = data.([]order.Detail)
+		wg.Done()
+	}()
+	require.NoError(t, b.wsProcessOrder(&wsOrderResponse))
+	wg.Wait()
+	require.True(t, ok)
+	require.Len(t, o, 1)
+	require.Equal(t, "c1956690-b731-4191-97c0-94b00422231b", o[0].OrderID)
+	require.Equal(t, "BTC_USDT", o[0].Pair.String())
+	require.Equal(t, order.Sell, o[0].Side)
+	require.Equal(t, order.Filled, o[0].Status)
+	require.Equal(t, 1.7, o[0].Amount)
+	require.Equal(t, 4.033, o[0].Price)
+	require.Equal(t, 4.24, o[0].AverageExecutedPrice)
+	require.Equal(t, 0., o[0].RemainingAmount)
+	require.Equal(t, asset.USDTMarginedFutures, o[0].AssetType)
```

Comment:
🐒  Also here; Can this not use the FixtureToDataHandler pattern?

Reply:
[removed and consolidated in](https://github.com/thrasher-corp/gocryptotrader/pull/1670/commits/49930d2190a74540225f1d89737ac66039fa6cc7)